data-prepper-core/src/main/java/org/opensearch/dataprepper/parser/MultiBufferDecorator.java

```diff
@@ -50,6 +50,12 @@ public Optional<Integer> getMaxRequestSize() {
         return  maxRequestSize.isPresent()  ? Optional.of(maxRequestSize.getAsInt()) : Optional.empty();
     }
 
+    @Override
+    public Optional<Integer> getOptimalRequestSize() {
+        OptionalInt optimalRequestSize = allBuffers.stream().filter(b -> b.getOptimalRequestSize().isPresent()).mapToInt(b -> (Integer)b.getOptimalRequestSize().get()).min();
```

Comment:
Please add some unit test cases. We should be sure to test both when this value is present and not present.